### ISSUE TITLE
"skimage.draw.draw" requires "matplotlib >=3.3'

### ISSUE THREAD

**Issue by ns-rse** [scikit-image/scikit-image#7281]
### Description:

I'm using skimage in some generate simple shapes for use in tests.

## `conftest.py`

```python
"""Fixtures for tests."""
import numpy as np
import numpy.typing as npt
import pytest
from skimage import draw


@pytest.fixture()
def square() -> npt.NDArray:
    """Return a 2D numpy array of a square."""
    square_array = np.zeros((6, 6), dtype=np.uint8)
    start = (1, 1)
    end = (5, 5)
    rr, cc = draw.rectangle_perimeter(start, end, shape=square_array.shape)
    square_array[rr, cc] = 1
    return square_array
```

## `test_shapes.py`

```python
"""Test the shapes module."""
import numpy.typing as npt
import pytest

from pytest_examples.shapes import summarise_shape


@pytest.mark.parametrize(
    ("shape", "area", "feret_diameter_max", "centroid"),
    [
        pytest.param("square", 11, 7.810249675906654, (1.3636363636363635, 1.3636363636363635), id="summary of square"),
        pytest.param("circle", 12, 5.385164807134504, (4, 4), id="summary of circle"),
    ],
)
def test_summarise_shape_get_fixture_value(
    shape: str,
    area: float,
    feret_diameter_max: float,
    centroid: tuple[int, int],
    request: npt.NDArray,
) -> None:
    """Test the summarisation of shapes."""
    shape_summary = summarise_shape(request.getfixturevalue(shape))
    assert shape_summary[0]["area"] == area
    assert shape_summary[0]["feret_diameter_max"] == feret_diameter_max
    assert shape_summary[0]["centroid"] == centroid
```

If I _don't_ have `matplotlib` as a `dependencies` in `pyproject.toml` then Matplotlib is not installed and the tests fail with the following (see [Action log](https://github.com/ns-rse/pytest-examples/actions/runs/7420792018/job/20192824071?pr=4)).

```bash
>           raise ImportError(msg + '"')
E           ImportError: "<function rectangle_perimeter at 0x7f37dfdd8b80>" in "skimage.draw.draw" requires "matplotlib >=3.3"
args       = ((1, 1), (5, 5))
kwargs     = {'shape': (6, 6)}
msg        = '"<function rectangle_perimeter at 0x7f37dfdd8b80>" in "skimage.draw.draw" requires "matplotlib >=3.3'
name       = 'matplotlib'
obj        = <function rectangle_perimeter at 0x7f37dfdd8b80>
version    = '>=3.3'

/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/skimage/_shared/version_requirements.py:141: ImportError
```

One solution of not adding `matplotlib` to the `dependencies` I found was to add `pytest-mpl` to the `optional.dependencies` under the `tests` set which are installed.

Tests pass since `pytest-mpl` pulls in `matplotlib` (see [completed action logs](https://github.com/ns-rse/pytest-examples/actions/runs/7421021315/job/20193539330?pr=4)).

### Version information:

This is running in CI, tests pass on my local environment as I have them installed.

From the [Action log](https://github.com/ns-rse/pytest-examples/actions/runs/7420792018/job/20192824071?pr=4)....

```Shell
Ubuntu 22.04.3-LTS
python-3.11.7
scikit-image-0.22.0
numpy-1.26.3
```

## Perhaps relevant?

I found #3624 which may be relevant, not sure.

I tried recreating a local environment and just installed `scikit-image` and not `matplotlib` to make a smaller reprex, but have `matplotlib` installed at the system level from where it is imported and don't want to start removing system level dependencies. Hope the clean environments in the GitHub runners are sufficient.

**Comment by lagru** [comment]
Hey, I am not sure I follow. `draw.rectangle_perimeter` requires the optional dependency `matplotlib` (for now) as you correctly identified. And you seem to have figured out a solution for this which is to specify `pytest-mpl` as a test dependency. You might also want to use `scikit-image[optional]` or `matplotlib` as test dependencies instead of `pytest-mpl`.

Is there something you would want to work differently?

**Comment by ns-rse** [comment]
:thinking: perhaps it would be as simple as adding a note to the documentation for `draw.rectangle_perimeter` to say that `scikit-image[optional]` is required.

I'd be happy to try that and make a PR if you want to assign the issue to me.

**Comment by mkcor** [comment]
The error message comes from
https://github.com/scikit-image/scikit-image/blob/0e35d8040d0b0bb409de41f3c0f024f941e09d8a/skimage/_shared/version_requirements.py#L140
I would think that it would be clear enough, admittedly, also assuming that the user would go through the [installation instructions](https://scikit-image.org/docs/stable/user_guide/install.html).

@ns-rse adding a note in the docstring for `draw.rectangle_perimeter` cannot hurt; it was done here
https://github.com/scikit-image/scikit-image/blob/0e35d8040d0b0bb409de41f3c0f024f941e09d8a/skimage/data/_fetchers.py#L1153-L1155
and there
https://github.com/scikit-image/scikit-image/blob/0e35d8040d0b0bb409de41f3c0f024f941e09d8a/skimage/data/_fetchers.py#L248-L251
It's just that it would be consistent, then, to include a similar note for *all* the `draw` functions(?).

**Comment by ns-rse** [comment]
Ok, I'll hold up my hand I didn't read all of the install instructions.

Perhaps rather than adding a comment to every `draw` function the linked PR might help those who don't make the _faux pas_ I have.

**Comment by ns-rse** [comment]
I think it wold be fine to closes this, especially in light of #7286 being merged which added a short sentence explaining the dependency.

**Comment by stefanv** [comment]
Thanks, Neil.
